### PR TITLE
update index and ibt

### PR DESCRIPTION
**Pull request by pete-ipor**

### CODE REVIEW

File: contracts/interfaces/IIporOracle.sol [hunk 1]
```diff
@@ -55,18 +54,22 @@ interface IIporOracle {
...
-    /// @dev INPUT_ARRAYS_LENGTH_MISMATCH if 'indexesToUpdate' array is empty.
-    /// @dev ASSET_NOT_SUPPORTED if an asset in 'indexesToUpdate' is not supported.
-    /// @dev WRONG_INDEX_TIMESTAMP if the provided timestamp is either older than the last update timestamp or greater
-    ///         than the current block timestamp.
+    /// @notice Updates the Indexes based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
+    /// The function accepts an array of the custom structure `UpdateIndexParams`.
+    /// The specifics of what this parameter actually does depends on how the function is implemented inside the contract.
+    /// The calldata specifies that the data should be stored in the calldata area which is read-only and cheaper to use than memory.
+    /// @param indexesToUpdate An array of `UpdateIndexParams` to be updated.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:62]: Lack of important  description that in structure UpdateIndexParams fields updateTimestamp and quasiIbtPrice are ignored.

File: contracts/interfaces/IIporOracle.sol [hunk 2]
```diff
@@ -55,18 +54,22 @@ interface IIporOracle {
...
     function updateIndexes(UpdateIndexParams[] calldata indexesToUpdate) external;
 
+    /// @notice Updates the Index and Quasi IBT price based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
+    /// The function accepts an array of the custom struct `IIporOracle.UpdateIndexParams`.
+    /// The specifics of what this parameter actually does depends on how the function is implemented inside the contract.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:68]: This line doesn't have sense. Lack of valuable information.

File: contracts/interfaces/IIporOracle.sol [hunk 3]
```diff
@@ -55,18 +54,22 @@ interface IIporOracle {
...
+    /// The calldata specifies that the data should be stored in the calldata area which is read-only and cheaper to use than memory.
+    /// @param indexesToUpdate An array of `IIporOracle.UpdateIndexParams` to be updated.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:70]: Lack of important  description that in structure UpdateIndexParams fields updateTimestamp and quasiIbtPrice are NOT ignored.

File: contracts/oracles/IporOracle.sol [hunk 4]
```diff
@@ -165,16 +158,25 @@ contract IporOracle is
         uint256 length = indexesToUpdate.length;
         require(length > 0, IporErrors.INPUT_ARRAYS_LENGTH_MISMATCH);
```
Comment (mario-ipor) [inline contracts/oracles/IporOracle.sol:159]: This validation is redundant - it had sense when we have two arrays in input param, now we have only one array with structure.

File: contracts/oracles/IporOracle.sol [hunk 5]
```diff
@@ -165,16 +158,25 @@ contract IporOracle is
...
+                ++i;
             }
+        }
+    }
+
+    function updateIndexesAndQuasiIbtPrice(
+        IIporOracle.UpdateIndexParams[] calldata indexesToUpdate
+    ) external override onlyUpdater whenNotPaused {
+        uint256 length = indexesToUpdate.length;
+        require(length > 0, IporErrors.INPUT_ARRAYS_LENGTH_MISMATCH);
```
Comment (mario-ipor) [inline contracts/oracles/IporOracle.sol:172]: This validation is redundant - it had sense when we have two arrays in input param, now we have only one array with structure.

File: contracts/oracles/IporOracle.sol
Comment (mario-ipor) [inline contracts/oracles/IporOracle.sol:1]: Setup version 2_002

**Review by mario-ipor — CHANGES_REQUESTED** [review]

File: contracts/interfaces/IIporOracle.sol (hunk 3 above)
Comment (pete-ipor) [reply contracts/interfaces/IIporOracle.sol:70]: we update all values so we don't need information about ignore or not

File: contracts/interfaces/IIporOracle.sol [hunk 6]
```diff
@@ -55,18 +54,20 @@ interface IIporOracle {
...
-    ///      requires the caller to be an authorized updater.
-    /// @param indexesToUpdate Array of 'UpdateIndexParams' containing asset address, new index value, update timestamp,
-    ///        and quasiIbtPrice (for '_stEth' asset).
-    /// @dev INPUT_ARRAYS_LENGTH_MISMATCH if 'indexesToUpdate' array is empty.
-    /// @dev ASSET_NOT_SUPPORTED if an asset in 'indexesToUpdate' is not supported.
-    /// @dev WRONG_INDEX_TIMESTAMP if the provided timestamp is either older than the last update timestamp or greater
-    ///         than the current block timestamp.
+    /// @notice Updates the Indexes based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
+    /// The function accepts an array of the custom structure `UpdateIndexParams`.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:59]: This is obvious so no need that in documentation. Code for this is self decriptive

File: contracts/interfaces/IIporOracle.sol [hunk 7]
```diff
@@ -55,18 +54,20 @@ interface IIporOracle {
...
+    /// The calldata specifies that the data should be stored in the calldata area which is read-only and cheaper to use than memory.
+    /// @param indexesToUpdate An array of `UpdateIndexParams` to be updated, in this function updateTimestamp and quasiIbtPrice are ignored.
     function updateIndexes(UpdateIndexParams[] calldata indexesToUpdate) external;
 
+    /// @notice Updates the Index and Quasi IBT price based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
+    /// The function accepts an array of the custom struct `IIporOracle.UpdateIndexParams`.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:66]: This is obvious so no need that in documentation. Code for this is self decriptive

File: contracts/interfaces/IIporOracle.sol [hunk 8]
```diff
@@ -55,18 +54,20 @@ interface IIporOracle {
...
-    ///         than the current block timestamp.
+    /// @notice Updates the Indexes based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
+    /// The function accepts an array of the custom structure `UpdateIndexParams`.
+    /// The calldata specifies that the data should be stored in the calldata area which is read-only and cheaper to use than memory.
+    /// @param indexesToUpdate An array of `UpdateIndexParams` to be updated, in this function updateTimestamp and quasiIbtPrice are ignored.
     function updateIndexes(UpdateIndexParams[] calldata indexesToUpdate) external;
 
+    /// @notice Updates the Index and Quasi IBT price based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:65]: This is obvious so no need that in documentation. Code for this is self decriptive

File: contracts/interfaces/IIporOracle.sol [hunk 9]
```diff
@@ -55,18 +54,20 @@ interface IIporOracle {
...
-    ///      Special handling is applied for the '_stEth' asset. Function execution is restricted during pause state and
-    ///      requires the caller to be an authorized updater.
-    /// @param indexesToUpdate Array of 'UpdateIndexParams' containing asset address, new index value, update timestamp,
-    ///        and quasiIbtPrice (for '_stEth' asset).
-    /// @dev INPUT_ARRAYS_LENGTH_MISMATCH if 'indexesToUpdate' array is empty.
-    /// @dev ASSET_NOT_SUPPORTED if an asset in 'indexesToUpdate' is not supported.
-    /// @dev WRONG_INDEX_TIMESTAMP if the provided timestamp is either older than the last update timestamp or greater
-    ///         than the current block timestamp.
+    /// @notice Updates the Indexes based on the provided parameters.
+    /// @dev This function is external and can be called from other contracts.
```
Comment (mario-ipor) [inline contracts/interfaces/IIporOracle.sol:58]: This is obvious so no need that in documentation. Code for this is self decriptive

**Review by mario-ipor — APPROVED** [review]

**Pull request merged**